samza-api/src/main/java/org/apache/samza/metrics/SamzaHistogram.java

```diff
@@ -53,6 +53,7 @@ public SamzaHistogram(MetricsRegistry registry, String group, String name, List<
 
   public void update(long value) {
     histogram.update(value);
+    percentiles.forEach(x -> updateGaugeValues(x));
```

Comment:
Looks like the gauge value is already updated when it is reported (line 76 Gauge#visit). Why do we need to update them on every new datapoint?

Reply:
good point. @prateekm is it possible that this is never called? We are seeing the values a 0, so I assumed this could be the issue.

Reply:
Also by the way this code is being called from Brooklin code in Brooklin cluster. Not sure if the visit is called

Reply:
@prateekm as discussed doing it in the getValue of the Histogram.